### PR TITLE
chore: bump zint to latest and remove patch

### LINKED ISSUE #10: Upgrade to latest zint
One thing to notice is that zint add "output to mem" option so we may not need to patch zint now

https://github.com/zint/zint/commit/98f86727cc73a02dba48aa27b49a76ead68b922c

### PR DESCRIPTION
**Pull request by Enter-tainer**
fix #10

### CODE REVIEW

**Comment by Enter-tainer** [comment]
Haven't figure out how to use the new mem file yet 😅

**Comment by Enter-tainer** [comment]
I might have figured out how it works. The code128 example is working now

**Comment by Enter-tainer** [comment]
image

For some reason rust code is still buggy

**Comment by Caellian** [comment]
Letting you know, I'm working on adding xtask to build the project. This will replace `build.sh` with a bunch of rust code that handles installing build requirements (WASI SDK, wasi-stub, wasm-opt) and packaging.

Noting this because it changes `/zint-wasm-sys/build.rs` but only the `env::set_var("CC", ...)` & `AR` lines. There should be minimal conflicts with this.

**Review by Caellian — COMMENTED** [review]
> For some reason typst binding is still buggy

That error is cause by some `unwrap` or `expect` somewhere.

File: zint-wasm-rs/src/symbol.rs [hunk 1]
```diff
@@ -127,15 +130,13 @@ impl Symbol {
...
-            let svg = CStr::from_ptr(svg_cstr)
-                .to_str()
-                .map_err(Error::InvalidResultSVG)?
-                .to_string();
-            free_svg_plot_string(svg_cstr);
-            (ZintResult::from(result as u32), svg)
+        let svg = unsafe {
+            let memfile = std::slice::from_raw_parts(self.memfile, self.memfile_size as usize);
+            let svg_str = CStr::from_bytes_with_nul(memfile)
+                .expect("failed to convert memfile to CStr")
```
[the remark this answers is not in the document]
Comment (Enter-tainer) [reply zint-wasm-rs/src/symbol.rs]: You are very correct! I(actually copilot) made a dumb mistake. I believe the memfile is never a CStr.

**Comment by Enter-tainer** [comment]
New wasm size:
```
-rwxr-xr-x 1 mgt mgt 850K Oct 17 23:42 typst-package/zint_typst_plugin.wasm*
```

**Comment by Enter-tainer** [comment]
I just checked the new zint.h. Besides the memory file option, there is not new constants and macros

**Review by Caellian — APPROVED** [review]
LGTM

**Pull request merged**